tests/e2e/test_rosacli_node_pool.go

```diff
@@ -806,4 +806,61 @@ var _ = Describe("Edit nodepool",
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("Editing versions is not supported, for upgrades please use 'rosa upgrade machinepool'"))
+
+				By("Try to edit the replicas of the machinepool with negative value")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--replicas", "-9")
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("The number of machine pool replicas needs to be a non-negative integer"))
+
+				By("Try to edit the machinepool with --min-replicas flag when autoscaling is disabled for the machinepool.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--min-replicas", "2")
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("Failed to get autoscaling or replicas: 'Autoscaling is not enabled on machine pool '%s'. can't set min or max replicas'", machinepoolName))
+
+				By("Try to edit the machinepool with --max-replicas flag when autoscaling is disabled for the machinepool.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--max-replicas", "5")
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("Failed to get autoscaling or replicas: 'Autoscaling is not enabled on machine pool '%s'. can't set min or max replicas'", machinepoolName))
+
+				By("Edit the machinepool to autoscaling mode.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--enable-autoscaling", "--min-replicas", "2", "--max-replicas", "6")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("Updated machine pool '%s' on hosted cluster '%s'", machinepoolName, clusterID))
+
+				By("Try to edit machinepool with negative min_replicas value.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--min-replicas", "-3")
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("The number of machine pool min-replicas needs to be greater than zero"))
+
+				By("Try to edit machinepool with --replicas flag when the autoscaling is enabled for the machinepool.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--replicas", "3")
+				Expect(err).To(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("Failed to get autoscaling or replicas: 'Autoscaling enabled on machine pool '%s'. can't set replicas'", machinepoolName))
+
+				By("Try to edit machinepool with just --autorepair flag.")
+				output, err = machinePoolService.EditMachinePool(clusterID, machinepoolName, "--autorepair=false")
```

Comment:
This step should be in the positive case(high) too otherwise it will not be run with 'Critical+high' priority in one job.